### ISSUE TITLE
Feature: from-path

### ISSUE THREAD

**Issue by cburgmer** [gga/json-path#4]
Hey Giles,

JsonPath is a perfect fit when finding values for an arbitrarily complex query. However, the key to that value is lost in the process.

More specifically, my users have a Spring Boot status page, and the health indicators are aggregated by key. Example status page:

```json
{"status":"UP","diskSpace":{"status":"UP","free":7537176576,"threshold":10485760}}
```
Naturally they want to display the health attached to the key "diskSpace".

I'm proposing the following API:
```clj
(json-path/from-path "$..[?(@.bar)]" [{:foo {:bar "Hello, world!"}}])
; => ([0 :foo])
(get-in [{:foo {:bar "Hello, world!"}}] [0 :foo])
; => {:bar "Hello, world!"}
```

```clj
(json-path/from-path "$.hello" {:hello "world"})
; => [:hello]
(get-in {:hello "world"} [:hello])
; => "world"
```

```clj
(json-path/from-path "$" {:hello "world"})
; => []
(get-in {:hello "world"} [])
; => {:hello "world"}
```
Thoughts?

**Comment by gga** [comment]
This is proposing a new interface to json-path that takes a JsonPath and converts it into a Clojure key path as suitable for the core `get-in` function, right?

That sounds like an excellent idea! If this were to land, then I'd deprecate `at-path` in favour of recommending `(comp get-in from-path)`.

Would `(comp flatten parse-path)` do the job?

**Comment by cburgmer** [comment]
Exactly, return a key path as understood by `get-in`. This is implemented now in PR #7.

Just using `parse-path` is not enough, as it only parses the JSON path query, it does not execute for matches. 
What I have done is extend the `walk` method to keep a record of the key path on the way when drilling down to the match. When the call finishes we hold both key and value for a given query. I found keeping hold of both key and value during the query execution makes for the most readable code.

As for `(comp get-in from-path)`: In the PR #7 I've deviated from the original plan for convenience/performance reasons. Rather than exposing `from-path`, the now proposed `query` will return both key and value for matches. I don't have a strong opinion on either, a smaller API (which you mention) might be easier to grasp for a new user, I don't know. On the other hand, throwing away the value in `from-path`, which the user then has to retrieve again via `get-in`, seems a bit like computational/memory waste.

**Comment by cburgmer** [comment]
I've merged the changes. I'll keep this open so we can discuss any API naming/refactoring.

**Comment by cburgmer** [comment]
With the latest changes we now have a `query` method, which will return one or several match records containing `:path` and `:value` keys. `at-path` is internally reduced to being a convenience method which will project the value.

**Comment by cburgmer** [comment]
Released as 1.0.0
